Classes/Core/Functional/Framework/Frontend/InternalRequest.php

```diff
@@ -61,13 +61,20 @@ function (array $data) {
 
     /**
      * @param string|null $uri URI for the request, if any.
+     * @param string $method method to use, GET is default.
+     * @param string|null $content body content to use, if any.
+     * @param array $headers Headers for the message, if any.
      */
-    public function __construct($uri = null) {
+    public function __construct($uri = null, $method = 'GET', string $content = null, array $headers = []) {
```

Comment:
how about we'll add an (optional) stream instead of a string?

Reply:
Whatever is not converted too many times firther down the road.

Reply:
Strangely moving the
```
        $body = new Stream('php://temp', 'wb+');
        if (null !== $content) {
            $body->write($content);
            $body->rewind();
        }
```

to the outer class and changing the type of the parameter broke everything. The body does not get passed along to the ServerRequestFactory

Reply:
> Strangely moving the
> 
> ```
>         $body = new Stream('php://temp', 'wb+');
>         if (null !== $content) {
>             $body->write($content);
>             $body->rewind();
>         }
> ```
> 
> to the outer class and changing the type of the parameter broke everything. The body does not get passed along to the ServerRequestFactory

Sorry, I give up on this.